lib/preview/shadowDevServer.js

```diff
@@ -0,0 +1,86 @@
+const net = require('net');
+const express = require('express');
+const { Router } = require('express');
+const cors = require("cors");
+
+const SHADOW_PORT = 1442;
+
+const startShadowDevServer = async (sessionInfo) => {
+  const portIsTaken = await new Promise((res, rej) => {
```

Comment:
Just a note for when you port this over to local-dev-lib. We have a port management util that may be useful https://github.com/HubSpot/hubspot-local-dev-lib/pull/42